### PR TITLE
feat: use t3-env to validate envs

### PR DESCRIPTION
**Pull request by G3root**
This PR aims to address breaking changes in self-hosted setups caused by renaming or adding new environment variables by implementing validation and establishing a single source of truth.

### CODE REVIEW

File: apps/web/src/env.mjs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  NEXT_PUBLIC_FEATURE_BILLING_ENABLED,
+
+  // APP
+  APP_VERSION,
+} = process.env;
+
+export const env = createEnv({
+  server: {
+    // AUTH
+    NEXTAUTH_URL: z.string().url(),
```
Comment (G3root) [inline apps/web/src/env.mjs:63]: environment variables from `.env.example` and some variable used in the webapp

File: packages/eslint-config/index.cjs [hunk 2]
```diff
@@ -68,4 +69,12 @@ module.exports = {
     // warnings so we can write appropriate interfaces and guards later.
     '@typescript-eslint/consistent-type-assertions': ['warn', { assertionStyle: 'never' }],
   },
+  overrides: [
+    {
+      files: ['apps/web/**/*.{tsx,ts}'],
+      rules: {
+        'no-process-env': 'error',
```
Comment (G3root) [inline packages/eslint-config/index.cjs:76]: restrict accessing `process.env` only for webapp

File: turbo.json [hunk 3]
```diff
@@ -72,6 +65,7 @@
     "POSTGRES_URL",
     "DATABASE_URL",
     "POSTGRES_PRISMA_URL",
-    "POSTGRES_URL_NON_POOLING"
+    "POSTGRES_URL_NON_POOLING",
+    "NEXT_PRIVATE_STRIPE_WEBHOOK_SECRET"
```
Comment (G3root) [inline turbo.json:69]: added  `NEXT_PRIVATE_STRIPE_WEBHOOK_SECRET` to globalEnv due to an eslint error

File: packages/eslint-config/index.cjs (hunk 2 above)
Comment (Mythie) [reply packages/eslint-config/index.cjs:76]: ooo

File: turbo.json (hunk 3 above)
Comment (Mythie) [reply turbo.json:69]: my bad

**Comment by Mythie** [comment]
Hey there, still considering this one sorry for the lack of movement.

**Comment by ElTimuro** [comment]
@G3root Thanks for contributing to Malfunction Mania, we really appreciate it.
With your contribution, you earned the limited edition Malfunction Mania "Glitchy" shirt. 
I messaged you on Twitter for details. 🫡

image

**Comment by Mythie** [comment]
Closing for now as it feels unlikely that we will adopt this in the near term. Will keep it noted for reference if we decide to come back to it in the future 💚

**Pull request closed without merging**